src/java.base/share/classes/java/util/WeakHashMap.java

```diff
@@ -447,6 +513,8 @@ Entry<K,V> getEntry(Object key) {
      *         {@code null} if there was no mapping for {@code key}.
      *         (A {@code null} return can also indicate that the map
      *         previously associated {@code null} with {@code key}.)
```

Comment:
Need to specify: A null return can also indicate the entry is discarded if the value policy is DISCARD and the key is a value object.

Reply:
A subtle point...  The WeakHashmap *may* contain an identity class instance that compares as `equals()` with the value object. In that case, the instance would be returned from `getEntry`.
Similarly, `put()` would return the previous entry (that was `equals`) to a value entry that is discarded.
This will need some more spec detail.